File: log/src/main/java/ua/at/tsvetkov/util/FragmentLifecycleLogger.java
package ua.at.tsvetkov.util;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;

class FragmentLifecycleLogger extends FragmentManager.FragmentLifecycleCallbacks {

   private String parent = "";
   private String info = null;

   @Override
   public void onFragmentAttached(@NonNull FragmentManager fm, @NonNull Fragment fr, @NonNull Context context) {
      super.onFragmentAttached(fm, fr, context);
      parent = context.getClass().getSimpleName();
      int backStackCount = fm.getBackStackEntryCount();
      info = Format.getFragmentsStackInfo(fm, "attached " + fr.getClass().getSimpleName(), backStackCount);
      android.util.Log.i(parent, Format.getFormattedMessage(info, "Fragments stack [" + backStackCount + "]").toString());
   }

   @Override
   public void onFragmentDetached(@NonNull FragmentManager fm, @NonNull Fragment fr) {
      super.onFragmentDetached(fm, fr);
      int backStackCount = fm.getBackStackEntryCount();
      info = Format.getFragmentsStackInfo(fm, "detached " + fr.getClass().getSimpleName(), backStackCount);
      android.util.Log.i(parent, Format.getFormattedMessage(info, "Fragments stack [" + backStackCount + "]").toString());
   }

}

File: lib/src/main/java/ua/at/tsvetkov/util/FragmentLifecycleLogger.kt
package ua.at.tsvetkov.util

import android.content.Context
import android.util.Log
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import ua.at.tsvetkov.util.Format.getFormattedMessage
import ua.at.tsvetkov.util.Format.getFragmentsStackInfo

internal class FragmentLifecycleLogger : FragmentManager.FragmentLifecycleCallbacks() {

    private var parent = ""
    private var info: String? = null

    override fun onFragmentAttached(fm: FragmentManager, fr: Fragment, context: Context) {
        super.onFragmentAttached(fm, fr, context)
        parent = context.javaClass.simpleName
        val backStackCount = fm.backStackEntryCount
        info = getFragmentsStackInfo(fm, "attached " + fr.javaClass.simpleName, backStackCount)
        Log.i(parent, getFormattedMessage(info!!, "Fragments stack [$backStackCount]").toString())
    }

    override fun onFragmentDetached(fm: FragmentManager, fr: Fragment) {
        super.onFragmentDetached(fm, fr)
        val backStackCount = fm.backStackEntryCount
        info = getFragmentsStackInfo(fm, "detached " + fr.javaClass.simpleName, backStackCount)
        Log.i(parent, getFormattedMessage(info!!, "Fragments stack [$backStackCount]").toString())
    }

}